pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -971,6 +973,7 @@ protected void handleConnect(CommandConnect connect) {
                 authRole = getBrokerService().getAuthenticationService().getAnonymousUserRole()
```

Comment:
This means that the proxy cannot use the anonymous role to connect to the broker, I don't think this is correct.

Usually, the proxy always enable the authentication feature, but we also should consider a case that the proxy doesn't enable the authentication feature.

Reply:
> This means that the proxy cannot use the anonymous role to connect to the broker

Yes, this is the design I am proposing we move towards. That is why I said this earlier in this PR:

> I propose that we consider a proxy connecting as the anonymous role as a misconfiguration.

> Usually, the proxy always enable the authentication feature, but we also should consider a case that the proxy doesn't enable the authentication feature.

In this case, the broker wouldn't enable authentication either. In the admin API, we only look for the original principal when the `authRole` is in the `proxyRoles` set. Given that the proxy has to have at least as much permission as the `originalPrincipal`, I do not see what the point of authentication would be if the anonymous role is also a proxy role.

Reply:
Your idea is very true, we also recommend enabling authentication anywhere.

We have a case in which we don't want to check the proxy authentication in the broker, only check the original authentication. Would you have any idea?

Right now, we set up `anonymousUserRole`, but we cannot get the original authentication, because of the broker jumped to `completeConnect` method, there is misses storing the original authentication.

Reply:
> We have a case in which we don't want to check the proxy authentication in the broker, only check the original authentication. Would you have any idea?

That context helps a lot, thanks! Is it possible to add a configuration to the proxy so that it forwards the original authentication data as the `authData` part of the `Connect` command? The broker wouldn't necessarily know the connection is from the proxy, but I think this should be fine.

I think this discussion probably relates to https://github.com/apache/pulsar/issues/19332. I have been thinking that our authentication state integration between proxy and broker is very complicated and doesn't appear to be working in all cases (see also https://github.com/apache/pulsar/issues/19291).

It seems like we could always drop the proxy auth data in cases where the authentication data can be forwarded.

Reply:
Sounds good, we need to improve the proxy codebase to find a workaround, this is going to take some time.